src/bp/ui-studio/src/web/views/FlowBuilder/diagram/index.tsx

```diff
@@ -193,10 +229,16 @@ export default class FlowBuilder extends Component<Props> {
     this.manager.sanitizeLinks()
     this.manager.cleanPortLinks()
 
-    if (!selectedNode && currentNode) {
+    if (!selectedNode) {
+      this.props.closeFlowNodeProps()
       this.props.switchFlowNode(null) // No node selected
     } else if (selectedNode && (!currentNode || selectedNode.id !== currentNode.id)) {
-      this.props.switchFlowNode(selectedNode.id) // Selected a new node
+      this.props.switchFlowNode(selectedNode.id)
+      if (!this.movingNode) {
+        this.props.openFlowNodeProps()
+      }
+    } else if (selectedNode && currentNode && selectedNode.id === currentNode.id && !this.movingNode) {
+      this.props.openFlowNodeProps()
```

Comment:
Small reminder to check the target of the selected element on the diagram engine to ensure it's a standard node, skill node or router node